core/src/edu/asu/c3simulator/screens/MarketingScreen.java

```diff
@@ -0,0 +1,368 @@
+	private Table createMainTable()
+	{
+		Table mainTable = new Table();
+		mainTable.setSize(0.80f * stage.getWidth(), 0.80f * stage.getHeight());
+		
+		Table leftSection = new Table();
+		Table rightSection = new Table();
+		
+		ScrollPane productGridScroll = new ScrollPane(getProductGrid(), skin);
+		companyPromotionButton = new TextButton("Promote Company", skin);
+		companyPromotionButton.addListener(new ClickListener() {
+			@Override
+			public void clicked(InputEvent event, float x, float y)
+			{
+				if (promoteCompany)
+					companyPromotionButton.setColor(1.0f, 1.0f, 1.0f, 1.0f);
+				else
+					companyPromotionButton.setColor(0.0f, 1.0f, 0.0f, 1.0f);
+				promoteCompany = !promoteCompany;
+				productChosen = false;
+				if (selectedProduct != null)
+					selectedProduct = null;
+				if (selectedProductLabel != null)
+				{
+					selectedProductLabel.setColor(1.0f, 1.0f, 1.0f, 1.0f);
+					selectedProductLabel = null;
+				}
+			}
+		});
+		
+		leftSection.add(productGridScroll).expand().fill().row();
+		leftSection.add(companyPromotionButton).center().space(10.0f);
+		
+		ScrollPane marketingGridScroll = new ScrollPane(getMarketingGrid(), skin);
+		TextButton marketinLabel = new TextButton("Select Marketing", skin);
+		marketinLabel.setDisabled(true);
+		marketinLabel.setColor(0.5f, 0.5f, 0.5f, 1.0f);
+		TextButton nextButton = new TextButton("Next", skin);
+		nextButton.addListener(new ClickListener() {
+			@Override
+			public void clicked(InputEvent event, float x, float y)
+			{
+				if (marketingMethodChosen && (productChosen || promoteCompany))
+				{
+					System.out.println("Moving to the next Step. "
+							+ selectedMarketingMethod
+							+ " will be used to market "
+							+ (productChosen ? selectedProduct.getName() : "the Company."));
+					game.setScreen(new MarketingWizard(game, MarketingScreen.this));
+				}
+			}
+		});
+		
+		rightSection.add(marketinLabel).expand().fill().row();
+		rightSection.add(marketingGridScroll).expand().fill().row();
+		rightSection.add(nextButton).right().space(10.0f);
+		
+		mainTable.add(leftSection).expand().fill();
+		mainTable.add(rightSection).expand().fill();
+		
+		return mainTable;
+	}
+	
+	/**
+	 * This method creates the the grid of products that the user can choose from and adds
+	 * them to a table.
+	 * 
+	 * @return The table that was created.
+	 */
+	private Table getProductGrid()
+	{
+		Table productGrid = new Table();
+		Product[] products = getProducts();
+		
+		for (int i = 0; i < products.length; i++)
+		{
+			Table newProduct = new Table();
+			Label productLabel = new Label(products[i].getName(), skin);
+			Image productIcon = products[i].getProductImage();
+			productIcon.addListener(new productIconListener(productLabel, products[i]));
+			
+			newProduct.add(productIcon).size(100f).row(); // TODO change to dynamic size
+			newProduct.add(productLabel);
+			
+			productGrid.add(newProduct).space(15);
+			
+			if ((i + 1) % 2 == 0 && i != products.length - 1)
```

Comment:
This looks like "if i is odd and not the last index"
check for oddness: (i & 1) 

Better yet, this loop could be cleaned up a bit by using an iterator, and using iterator.hasNext() to see if it's the last element.